### DISCUSSION TITLE
ImageMagick in CMYK makes a black channel in multiple colors

### DISCUSSION THREAD

**Discussion by xa-xaker** [ImageMagick/ImageMagick#3581] [Help]
After running a simple command `convert source.pdf output.pdf`
the image output.pdf changes the CMYK structure.
In source.pdf the black channel was in one color, and in output.pdf it was in multiple colors.

in Photoshop, you can demonstrate this as follows:

source.pdf. If you turn off the black channel, then all black color in the image disappears.
http://joxi.ru/vAWl4ajcOJ5kdr

output.pdf. And here, if you turn off the black channel, all the text remains in other colors.
http://joxi.ru/KAgYk4jFKpyE6m

P.S. Ignore the resolution of the output.pdf image. This does not apply to the topic.

I would be grateful for your help!

**Comment by snibgo** [comment]
I don't understand the problem.

You link to two images. What are they? Are they screen captures of some program that is displaying the PDF files? If so, then please post links to the actual PDF files.

> After running a simple command convert source.pdf output.pdf
the image output.pdf changes the CMYK structure.
In source.pdf the black channel was in one color, and in output.pdf it was in multiple colors.

I don't understand. IM will rasterize each PDF, which converts the image to pixels. Then it writes that raster image to a PDF. The output is almost guaranteed to be different to the input.

**Comment by xa-xaker** [comment]
> You link to two images. What are they? Are they screen captures of some program that is displaying the PDF files? If so, then please post links to the actual PDF files.

Yes, these are screens from Photoshop. To demonstrate the problem. 

Here are the links to the files themselves: 
http://wicc.net.ua/source.pdf
http://wicc.net.ua/output.pdf

I say it again. The problem is not the resolution, but the CMYK structure

**Comment by snibgo** [comment]
When I view source.pdf and output.pdf in Adobe Acrobat Reader DC, they clearly have different resolutions. Other than that, they look the same.

**Reply by xa-xaker** [reply]
Resolution doesn't matter. They also differ in the CMYK structure. I showed in the screenshots from Photoshop how it looks

**Reply by fmw42** [reply]
Perhaps your Imagick or Imagemagick are buggy?

**Comment by snibgo** [comment]
When IM rasterizes source.pdf at density 300...
```
magick -verbose -density 600 -colorspace sRGB source2.pdf[0] x1.tiff
```
... we find the letters are not black. They are typically sRGB(13.7255%, 12.1569%, 12.549%). So they will have non-zero but small values in the C, M and Y channels.

I don't use Photoshop, and can't comment on that.

**Comment by fmw42** [comment]
Your output.pdf looks the same except for the quality. When you go from PDF to PDF, Imagemagick rasterizes the input PDF and saves a raster image in the output PDF. It is no longer vector text, but raster text, which has been smeared by the coarse rasterization into new colors. You can get a better quality result as follows doing supersampling at a factor of 8:

`convert -density 576 source.pdf -resize 12.5% -density 72 new.pdf`

where 72*8 = 576 and 100/8 = 12.5
